part2/stages/Functions/install-main

```diff
@@ -486,6 +523,55 @@ mount_dom0()
     fi
 }
 
+umount_upgrade_compat()
+{
+    do_umount ${UPGRADE_MOUNT}/boot/system &&
+    if grep -qs "^securityfs " /proc/mounts ; then
+        do_umount ${UPGRADE_MOUNT}/sys/kernel/security
+    fi &&
+    if grep -qs "^selinuxfs " /proc/mounts ; then
+        do_umount ${UPGRADE_MOUNT}/sys/fs/selinux
+    fi &&
+    do_umount ${UPGRADE_MOUNT}/proc &&
+    do_umount ${UPGRADE_MOUNT}/sys &&
+    do_umount ${UPGRADE_MOUNT}/dev &&
+    do_umount ${UPGRADE_MOUNT}/tmp &&
+    do_umount ${UPGRADE_MOUNT}/boot &&
+    do_umount ${UPGRADE_MOUNT}/config &&
+    do_umount ${UPGRADE_MOUNT}/etc/selinux/xc_policy/contexts/files &&
+    do_umount ${UPGRADE_MOUNT}/etc/xen/xenrefpolicy/policy &&
+    do_umount ${UPGRADE_MOUNT}
+}
+
+mount_upgrade_compat()
+{
+    local tmpfsopts="size=64M"
+    mkdir -p ${UPGRADE_MOUNT}
+
+    if selinux_enforced ; then
+        tmpfsopts="$tmpfsopts,rootcontext=system_u:object_r:tmp_t:s0"
+    fi
+
+    do_mount -o ro /dev/xenclient/upgradecompat ${UPGRADE_MOUNT} &&
+    do_mount -o bind /proc ${UPGRADE_MOUNT}/proc &&
+    do_mount -o bind /sys ${UPGRADE_MOUNT}/sys &&
+    do_mount -o bind /dev ${UPGRADE_MOUNT}/dev &&
+    do_mount -t tmpfs -o $tmpfsopts tmpfs ${UPGRADE_MOUNT}/tmp &&
+    do_mount -o bind ${DOM0_MOUNT}/boot ${UPGRADE_MOUNT}/boot &&
+    do_mount -o bind ${DOM0_MOUNT}/config ${UPGRADE_MOUNT}/config &&
+    do_mount -o bind ${DOM0_MOUNT}/etc/selinux/xc_policy/contexts/files/ ${UPGRADE_MOUNT}/etc/selinux/xc_policy/contexts/files/ &&
+    do_mount -o bind ${DOM0_MOUNT}/etc/xen/xenrefpolicy/policy/ ${UPGRADE_MOUNT}/etc/xen/xenrefpolicy/policy/ &&
```

Comment:
I don't think you need xenrefpolicy since it's the hypervisor Flask policy and not used by selinux.

Reply:
It is one of the objects measured in PCR8 for UEFI SRTM and the upgrade-compat image does not ship it?

Reply:
seal-system looks for policy.24 when forward sealing, line 98 in pcr8_objs

Reply:
Thanks. Yeah, I want only thinking about setfiles. Sorry for the noise